src/gui/gui-input.c

```diff
@@ -948,6 +948,96 @@ gui_input_delete_next_word (struct t_gui_buffer *buffer)
+                                         buffer->input_buffer_pos);
+
+        start_of_line = (char *)utf8_start_of_line (buffer->input_buffer, start);
+        if (start_of_line == start) {
+            start_of_line = (char *)utf8_prev_char (buffer->input_buffer, start_of_line);
+            start_of_line = (char *)utf8_start_of_line (buffer->input_buffer, start_of_line);
+        }
+        if (!start_of_line) {
+            start_of_line = buffer->input_buffer;
+        }
+
+        size_deleted = start - start_of_line;
+        length_deleted = utf8_strnlen (start_of_line, size_deleted);
+        gui_input_clipboard_copy (start_of_line, size_deleted);
+
+        memmove (start_of_line, start, strlen (start));
+
+        if (gui_input_optimize_size (
+                buffer,
+                buffer->input_buffer_size - size_deleted,
+                buffer->input_buffer_length - length_deleted))
+        {
+            buffer->input_buffer[buffer->input_buffer_size] = '\0';
+            buffer->input_buffer_pos = utf8_pos (buffer->input_buffer,
+                                                 start_of_line - buffer->input_buffer);
+        }
+        gui_input_text_changed_modifier_and_signal (buffer,
+                                                    1, /* save undo */
+                                                    1); /* stop completion */
+    }
+}
+
+/*
+ * Deletes all from cursor pos to end of line (default key: ctrl-K).
+ */
+
+void
+gui_input_delete_end_of_line (struct t_gui_buffer *buffer)
+{
+    int length_deleted, size_deleted;
+    char *start, *end_of_line;
+
+    if (buffer->input)
+    {
+        gui_buffer_undo_snap (buffer);
+        start = (char *)utf8_add_offset (buffer->input_buffer,
+                                         buffer->input_buffer_pos);
+
+        if (start[0] && start[0] == '\n')
+            end_of_line = (char *)utf8_next_char (start);
+        else
+            end_of_line = start;
+        end_of_line = (char *)utf8_end_of_line (end_of_line);
+
+        size_deleted = end_of_line - start;
+        length_deleted = utf8_strnlen (start, size_deleted);
+        gui_input_clipboard_copy (start, size_deleted);
+
+        memmove (start, end_of_line, strlen (end_of_line));
+
+        if (gui_input_optimize_size (
+                buffer,
+                buffer->input_buffer_size - size_deleted,
+                buffer->input_buffer_length - length_deleted))
+        {
+            buffer->input_buffer[buffer->input_buffer_size] = '\0';
+            buffer->input_buffer_pos = utf8_pos (buffer->input_buffer,
+                                                 start - buffer->input_buffer);
+        }
+
+        gui_input_text_changed_modifier_and_signal (buffer,
+                                                    1, /* save undo */
+                                                    1); /* stop completion */
+    }
+}
+
+/*
+ * Deletes all from cursor pos to beginning of input (default key: TODO).
```

Comment:
I'm not sure which keys would be suitable to use by default for `delete_beginning_of_input` and `delete_end_of_input`. Any suggestions?

Reply:
I went with `meta` plus the same keys as for line (since ctrl and shift can't be used together). I.e. `meta-ctrl-U` for `/input delete_beginning_of_input` and `meta-ctrl-K` for `/input delete_end_of_input`.